proposal.rst

```diff
@@ -267,8 +304,191 @@ Project Timeline
+        And the documentation would be providied in godoc official website 
+        and xapian.org.
+
+    3.  Provide the go get feature for using xapian bindings in go with 
+        proper guide lines regarding whether the xapian core is installed
+        or not.
+
+Plan :
+      * Go is Statically-typed language and each variable can only be of one type.
+        
+        Example :
+        /*
+        type Integer int
+            
+        var VarA Integer =10;
+        var VarB int = 20
+
+        func example(x int) */
+
+        This produces an error when  VarA is passed to func.
+        Even though internally Integer is an int but both are different types.
+        Code and error here (https://pastebin.com/bvz5QLZJ).
+
+        The way swig wraps the enums is not that natural and there should type conversions before passing to appropriate 
+        function for proper functioning. (https://pastebin.com/X8K1q9Rh)
+
+        Rewrap the go code(which does conversions before passing to the that particular function)
+        in to wrapper for enums using swig %insert(go_wrapper).(https://github.com/srinivasyadav18/xapian-gsoc-plan/blob/master/example.i#L51)
+      
+      * Go does not support constructors but this can be done with an extra helper function that takes slice of interfaces
+        which swig does by default during the wrapping, but little extra code need to be added for constructors and
+        functions when overloaded as they take slice of interfaces in ellipse syntax.
+        interface{} in golang means any type. Slice of interfaces mean collection of interfaces(resizable array).
+        Go supports variable number of arguments of different type to functions as func myfun(a ...interface{}) which is used during 
+        constructor and function overloading.
+         
+      * Go supports Iterators by natural syntax using channels and convential methods such as Iter.Next().
+        
+        1. Using channels one could use for-range construct.
+
+        for i := range doc.terms(){
+          i.GetTerm() // methods to get term from the iterator at that position.
+        }
+	(sample code how channels are used for iteration with for-range construct https://github.com/srinivasyadav18/xapian-gsoc-plan/blob/master/go.i#L43)
+        2. Using methods such as Iter.Next() as used in Go lang standard library (Container List https://golang.org/pkg/container/list/).
+         
+        Both standard method /* for iter.Next(){ ... code } */ and /* for-range construct would be made available for user
+        (these are just like auto-ranged based loops in c++) */
+         
+        Go support multiple return values , therefore rewrapping the interfaces which return iterators to both 
+        begin and end iterators in one function call as below.
+        /* begin,end := doc.Terms() */ 
+
+      * Go supports errors as return values . A language like c++ have try catch block Go has three constructs for dealing
```

Comment:
You talk about handling exceptions using panic/defer/recover, but is this the right approach? Should our exceptions instead be converted into return values? Can you write some example golang showing how to handle a common case? (The most obvious is probably when the `Database()` raises an exception if you tell it to open a database that doesn't exist.)

Reply:
> You talk about handling exceptions using panic/defer/recover, but is this the right approach? Should our exceptions instead be converted into return values? Can you write some example golang showing how to handle a common case? (The most obvious is probably when the `Database()` raises an exception if you tell it to open a database that doesn't exist.)

Hey I have put the example case for Database already...please check, it catches the error as return value during call of the constructer. I have followed the pattern that go  standard library follows.